tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c

```diff
@@ -0,0 +1,302 @@
+	expect_single_call_bt_unpair(oldest_params_vector->id, oldest_params_vector->addr);
+
+	return 0;
+}
+
+/* This call back will be called when bt_unpair() is called with the same parameters
+ * It can be used when a call to bt_unpair() isn't expected
+ */
+static int bt_unpair_unreachable_custom_fake(uint8_t id, const bt_addr_le_t *addr)
+{
+	ARG_UNUSED(id);
+	ARG_UNUSED(addr);
+	zassert_unreachable("Unexpected call to 'bt_unpair()' occurred");
+	return 0;
+}
+
+/* This call back will be called when bt_conn_foreach() is called with the same parameters
+ * It will act as if there are some active connections
+ */
+static void bt_conn_foreach_custom_fake_key_slot_0_in_use(int type, bt_conn_foreach_cb func,
+							  void *data)
+{
+	struct bt_conn conn;
+
+	/* This will make the effect as if there is a disconnection */
+	conn.state = BT_CONN_DISCONNECTED;
+	conn.id = 0x9E;
+	func(&conn, data);
+
+	/* This will make the effect as if there is a connection with no key */
+	conn.state = BT_CONN_CONNECTED;
+	conn.id = 0xFF;
+	bt_addr_le_copy(&conn.le.dst, (const bt_addr_le_t *)BT_ADDR_LE_ANY);
+	bt_conn_get_dst_fake.return_val = &conn.le.dst;
+	func(&conn, data);
+
+	/* This will make the effect as if key at slot 0 is in use with a connection */
+	conn.state = BT_CONN_CONNECTED;
+	conn.id = testing_id_addr_pair_lut[0].id;
+	bt_addr_le_copy(&conn.le.dst, testing_id_addr_pair_lut[0].addr);
+	bt_conn_get_dst_fake.return_val = &conn.le.dst;
+	func(&conn, data);
+}
+
+/* This call back will be called when bt_conn_foreach() is called with the same parameters
+ * It will act as if all keys are in used with active connections
+ */
+static void bt_conn_foreach_custom_fake_all_keys_in_use(int type, bt_conn_foreach_cb func,
+							void *data)
+{
+	struct bt_conn conn;
+
+	/* This will make the effect as if key at slot 'x' is in use with a connection */
+	for (size_t i = 0; i < ARRAY_SIZE(testing_id_addr_pair_lut); i++) {
+		conn.state = BT_CONN_CONNECTED;
+		conn.id = testing_id_addr_pair_lut[i].id;
+		bt_addr_le_copy(&conn.le.dst, testing_id_addr_pair_lut[i].addr);
+		bt_conn_get_dst_fake.return_val = &conn.le.dst;
+		func(&conn, data);
+	}
+}
+
+/* This call back will be called when bt_conn_foreach() is called with the same parameters
+ * It will act as if all keys are in used with active connections
+ */
+static void bt_conn_foreach_custom_fake_no_keys_in_use(int type, bt_conn_foreach_cb func,
+							void *data)
+{
+	struct bt_conn conn;
+
+	/* This will make the effect as if key at slot 'x' is in use with a connection */
+	for (size_t i = 0; i < ARRAY_SIZE(testing_id_addr_pair_lut); i++) {
+		conn.state = BT_CONN_DISCONNECTED;
+		conn.id = testing_id_addr_pair_lut[i].id;
+		bt_addr_le_copy(&conn.le.dst, testing_id_addr_pair_lut[i].addr);
+		bt_conn_get_dst_fake.return_val = &conn.le.dst;
+		func(&conn, data);
+	}
+}
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list is full
```

Comment:
`Add a new (ID, Address) pair while the keys pool is full and all the keys are in use.`
Same for the next testcase with `and all the keys aren't in use`.